deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/layers/recurrent/LSTMHelpers.java

```diff
@@ -168,6 +170,15 @@ static public FwdPassReturn activateHelper(final Layer layer, final NeuralNetCon
             prevOutputActivations = Nd4j.zeros(new int[] {miniBatchSize, hiddenLayerSize});
         }
 
+        if (helper != null && Nd4j.dataType() != DataBuffer.Type.HALF) {
```

Comment:
Sure, I'd be happy with that - my criteria is basically (a) works seamlessly when we can't use the helper, and (b) the user gets a clear log message (but only once per network, or at most once per layer). Which the null+log approach can satisfy.